fendermint/app/src/app.rs

```diff
@@ -562,6 +595,14 @@ where
             .await
             .context("end failed")?;
 
+        if let Some(snapshot) = &self.snapshot {
+            if snapshot.should_run(request.height as BlockHeight) {
+                let (state_params, block_height) =
+                    self.state_params_at_height(Height::try_from(request.height)?)?;
```

Comment:
I don't think at this point the state parameters for `request.height` are already saved to history, so this will most likely end up returning the last committed state. 

I think a better place to kick off snapshotting is in the `commit` method, after the state has been saved to the database. Technically in `end_block` Tendermint hasn't told us to save the changes yet, so we should not be snapshotting just yet.

In `commit` these parameters are available without lookup as well.

Reply:
Agree with @aakoshh, we should trigger the snapshot either when we commit the state, or at the beginning of the next block (I personally think that the former makes more sense).